src/libraries/System.Memory/tests/Span/EnumerateRunePositions.cs

```diff
@@ -0,0 +1,248 @@
+        [InlineData("\0")]
+        public static void RunePosition_UnitTest(string runeStr)
+        {
+            Rune.DecodeFromUtf16(runeStr, out Rune rune, out _);
+
+            RunePosition runePosition1 = new RunePosition(rune, 0, rune.Utf16SequenceLength, false);
+            RunePosition runePosition2 = new RunePosition(rune, 5, rune.Utf16SequenceLength, true);
+            RunePosition runePosition3 = new RunePosition(rune, 100, 5, true);
+
+            foreach (var startIndex in new [] {0, 5, 100})
+            {
+                foreach (var wasReplaced in new[] { true, false })
+                {
+                    RunePosition runePosition = new RunePosition(rune, startIndex, rune.Utf16SequenceLength, wasReplaced);
+                    RunePosition_TestProps(rune, startIndex, rune.Utf16SequenceLength, wasReplaced, runePosition);
+
+                    RunePosition_TestEquals(default, runePosition);
+                    RunePosition_TestEquals(runePosition1, runePosition);
+                    RunePosition_TestEquals(runePosition2, runePosition);
+                    RunePosition_TestEquals(runePosition3, runePosition);
+                    RunePosition_TestDeconstruct(runePosition);
+                }
+            }
+        }
+
+        [Fact]
+        public static void EnumerateRunePositions_Empty()
+        {
+            {
+                RunePosition.Utf16Enumerator enumerator = RunePosition.EnumerateUtf16(ReadOnlySpan<char>.Empty).GetEnumerator();
+                Assert.Equal(default, enumerator.Current);
+                Assert.False(RunePosition.EnumerateUtf16(ReadOnlySpan<char>.Empty).GetEnumerator().MoveNext());
+                Assert.Equal(default, enumerator.Current);
+            }
+            //Assert.False(RunePosition.EnumerateUtf8(ReadOnlySpan<char>.Empty).GetEnumerator().MoveNext());
+        }
+
+        [Theory]
+        [InlineData(new char[0], new int[0])] // empty
+        [InlineData(new char[] { 'x', 'y', 'z' }, new int[] { 'x', 'y', 'z' })]
+        [InlineData(new char[] { 'x', '\uD86D', '\uDF54', 'y' }, new int[] { 'x', 0x2B754, 'y' })] // valid surrogate pair
+        [InlineData(new char[] { 'x', '\uD86D', 'y' }, new int[] { 'x', 0xFFFD, 'y' })] // standalone high surrogate
+        [InlineData(new char[] { 'x', '\uDF54', 'y' }, new int[] { 'x', 0xFFFD, 'y' })] // standalone low surrogate
+        [InlineData(new char[] { 'x', '\uD86D' }, new int[] { 'x', 0xFFFD })] // standalone high surrogate at end of string
+        [InlineData(new char[] { 'x', '\uDF54' }, new int[] { 'x', 0xFFFD })] // standalone low surrogate at end of string
+        [InlineData(new char[] { 'x', '\uD86D', '\uD86D', 'y' }, new int[] { 'x', 0xFFFD, 0xFFFD, 'y' })] // two high surrogates should be two replacement chars
+        [InlineData(new char[] { 'x', '\uFFFD', 'y' }, new int[] { 'x', 0xFFFD, 'y' })] // literal U+FFFD
+        public static void EnumerateRunePositions_Battery16(char[] chars, int[] expected)
```

Comment:
Since all of these inputs were copied from other Rune tests, and since this test relies on the proper functionality of `DecodeFromUtf16` (which is already tested), you don't actually need the _expected_ parameter here. Your call to `RunePosition_TestProps` is all that is needed, and you can skip keeping track of the _enumeratedValues_ local.

(Same comment in UTF8 test.)